src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -52,123 +37,141 @@
     }
   }
 
   public void unmute() {
-    if (muted) {
-      clickElement(String.format(SPEAKER_SELECTOR_FORMAT, slot));
-      muted = false;
+    if (isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
     }
   }
 
-  public void pause() {
-    if (playing) {
-      clickElement(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
-      playing = false;
-    }
+  public void togglePause() {
+    wait.forElementClickable(pauseOverlaySelector).click();
   }
 
   public void play() {
-    if (!playing) {
-      clickElement(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
-      playing = true;
+    if (isPausedWithOverlay()) {
+      togglePause();
+    } else if( !this.mobile ) {
+      clickOnArea(2);
+    } else {
+      clickOnArea(4);
     }
   }
 
-  public void playVuapVideo() {
-    if (playing) {
-      clickOnClickArea2();
-      playing = true;
-    }
+  private boolean isPausedWithOverlay() {
+    return isVideoPaused() && driver.findElement(pauseOverlaySelector).isDisplayed();
   }
 
-  public void replay() {
-      clickElement(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
-      muted = false;
-      playing = true;
+  private Boolean isVideoPaused() {
+    return isDesktop() ? isDesktopVideoPaused() : isMobileVideoPaused();
   }
 
-  public void close() {
-    findCloseButton().click();
+  private boolean isDesktop() {
+    return hasDesktopVideoElement();
+  }
+
+  private Boolean isMobileVideoPaused() {
+    return Boolean.valueOf(driver.findElement(getVideoSelector()).getAttribute("paused"));
   }
 
-  public void clickOnClickArea2() {
-    clickOnAdClickArea(AD_TNG_CLICK_AREA_2_SELECTOR);
+  private Boolean isDesktopVideoPaused() {
+    return Boolean.valueOf(usingImaBridge(webDriver -> {
+      final JavascriptActions js = new JavascriptActions();
+
+      return js.execute("document.querySelector('video') && document.querySelector('video').paused").toString();
+    }));
   }
 
-  public void clickOnClickArea4() {
-    clickOnAdClickArea(AD_TNG_CLICK_AREA_4_SELECTOR);
+  public void replay() {
+    wait.forElementVisible(replayOverlaySelector).click();
   }
 
-  public void clickOnAdImageResolvedState() {
-    clickOnAdClickArea(AD_RESOLVED_STATE_IMAGE_SELECTOR);
+  public void close() {
+    wait.forElementClickable(closeButtonSelector).click();
   }
 
-  private void clickOnAdClickArea(String clickAreaSelector) {
-    usingAdFrame(() -> wait.forElementClickable(By.cssSelector(clickAreaSelector)).click());
+  public void closeWithJS() {
+    clickByJS(closeButtonSelector);
   }
 
-  public double getCurrentTime() {
-    final String currentTime = usingVideoContext(video -> video.getAttribute("currentTime"));
-    return Double.parseDouble(currentTime);
+  private void clickByJS(By selector) {
+    new JavascriptActions(driver).execute("arguments[0].click();", driver.findElement(selector));
+  }
+
+  private void clickElementInsideAd(By selector) {
+    usingAdFrame(() -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove useless curly braces around statement (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1602)